code/modules/projectiles/gun_system.dm

```diff
@@ -458,6 +459,8 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		return
 
 	if(!istype(A, /obj/screen))
+		if(fire_animation)
+			flick("[fire_animation]", src)
 		Fire(A, user, params) //Otherwise, fire normally.
```

Comment:
This would be better suited inside the Fire proc

Reply:
where exactly part of the fire() proc you think it would be suited at to be moved ?

Reply:
Where you put it is good